scripts/get_tau_pet.py

```diff
@@ -1,94 +1,121 @@
-import numpy as np
 import pandas as pd
+import numpy as np
 
-'''
-We have a phenotypic spreadsheet with multiple visit dates per participant. We also have a spreadsheet with multiple
-tau pet values for (some) participants. The goal is to find each participant's baseline visit data from the phenotypic
-file, and then find a tau pet value that was collected within one year.
-'''
-
-# load baseline phenotypic data
-pheno = pd.read_csv('/Users/natashaclarke/Documents/SIMEXP/cwas_neuro_sandbox/adni_cwas/adni_spreadsheet.csv', index_col=0, header=0)
-
-# keep only the variables of interest
-pheno = pheno.filter(['Subject ID','Phase','Sex','Research Group', 'Visit','Study Date','Age'], axis=1)
-
-# convert 'study date' to 'session' in datetime format, to match other spreadsheets
-pheno['session'] = pd.to_datetime(pheno['Study Date'])
+from pathlib import Path
 
-# pull out only the subject id and asign it to the index
-pheno_subj = []
-for i in pheno['Subject ID']:
-    subj = i.split('_')[2].lstrip("0") # remove leading zeros since it won't match ADNI IDs
-    pheno_subj.append(subj)
-    
-pheno.index = pheno_subj
-pheno.rename_axis('RID',inplace=True)
-pheno.index = pheno.index.astype('int64')
+# load data
+adni_data = Path(__file__).parent.parent / 'data' / 'adni_spreadsheet.csv'
+tau_data = Path(__file__).parent.parent / 'data' / 'UCBERKELEYAV1451_04_26_22.csv'
 
-# separate patients and controls, because (in theory) we can use any control visit as baseline, but
-# for patients we want their actual baseline data
-ps = ['AD', 'EMCI', 'LMCI', 'MCI', 'SMC']
-pdf = pheno[pheno['Research Group'].isin(ps)] # df of patient diagnoses
+# functions
+def get_baselines(file):
 
-cdf = pheno.loc[pheno['Research Group'] == 'CN'] # df of control diagnoses
+    ''' Load multiple visits for adni participants and return a df with only baselines '''
 
-# I think these visits are acceptable as baseline data, i.e. actual baseline +/-3 months, excluding
-# any initial visits where patient continued from a previous phase
-p_visits = ['ADNI Screening','ADNI2 Month 3 MRI-New Pt', 'ADNI2 Screening MRI-New Pt', 
-                   'ADNIGO Month 3 MRI','ADNIGO Screening MRI']
+    # load baseline phenotypic data
+    pheno = pd.read_csv(file, index_col=0, header=0)
 
-pdf_bl = pdf[pdf['Visit'].isin(p_visits)]
+    # keep only the variables of interest
+    pheno = pheno.filter(['Subject ID','Phase','Sex','Research Group', 'Visit','Study Date','Age'], axis=1)
 
-# rejoin the patients to the controls
-new_df = pd.concat([cdf,pdf_bl])
+    # convert 'study date' to 'session' in datetime format, to match other spreadsheets
+    pheno['session'] = pd.to_datetime(pheno['Study Date'])
```

Comment:
One thing to make the code easier to follow - group the same type of task / information together. I will move this to just before you create the upper and lower bound where you handle time related info together.